### DISCUSSION TITLE
Where can I find errno list per micropython port?

### DISCUSSION THREAD

**Discussion by TheFern2** [micropython/micropython#11359] [General]
I have a try/catch for a socket connection in a python3 library, [docs](https://docs.micropython.org/en/latest/library/socket.html) say socket.error or gaierror are not available and instead OSError is thrown. socket.error did give a few different errors based on the error we have a dictionary to return different messages, how could I handle that in micropython?

I was trying to use errno.ECOMM and EIO but those are giving me an error that they don't exist. If the connection is bad I get a EINPROGRESS all the time.

> This module provides access to symbolic error codes for [OSError](https://docs.micropython.org/en/latest/library/builtins.html#OSError) exception. A particular inventory of codes depends on [MicroPython port](https://docs.micropython.org/en/latest/reference/glossary.html#term-MicroPython-port).

**Comment by dlech** [comment]
If you search the code for [MICROPY_PY_UERRNO_LIST](https://github.com/search?q=repo%3Amicropython%2Fmicropython%20MICROPY_PY_UERRNO_LIST&type=code), you can see. It looks like all ports other than cc3200 use the defaults.

**Reply by TheFern2** [reply]
Thank you!
